components/dashboard/src/settings/Billing.tsx

```diff
@@ -4,15 +4,37 @@
  * See License-AGPL.txt in the project root for license information.
  */
 
+import { useContext } from "react";
 import { PageWithSettingsSubMenu } from "./PageWithSettingsSubMenu";
 import { BillingAccountSelector } from "../components/BillingAccountSelector";
+import { UserContext } from "../user-context";
+import UsageBasedBillingConfig from "../components/UsageBasedBillingConfig";
+import { AttributionId } from "@gitpod/gitpod-protocol/lib/attribution";
 
 export default function Billing() {
+    const { user } = useContext(UserContext);
+
+    if (!user) {
+        return <></>;
```

Comment:
Same here: Returning `<></>` early when something is still loading prevents showing a spinner (or anything at all), and thus makes the UI look broken temporarily.

How about, instead, using `user?.id || ""` everywhere a string ID is needed, and setting `showSpinner={!user}` below?

Reply:
Thanks; done as you suggest.